src/main/java/com/datalogics/pdf/security/LunaHsmManager.java

```diff
@@ -71,8 +71,10 @@ public void hsmLogin(final HsmLoginParameters parms) {
 
         if (slotManager.isLoggedIn()) {
             state = ConnectionState.CONNECTED;
+            loadKeyStore();
+        } else {
+            throw new IllegalArgumentException("Could not log into the Luna HSM");
```

Comment:
That's a tricky one, because the specific behavior I was trying to prevent with this *is* due to a bad argument: a bad token/slot name. However, that's not the *only* reason the log in might fail, and there's not an obvious way to even distinguish among a bad slot name and other causes, let alone report it.